jq_properties_parser.go

```diff
@@ -0,0 +1,104 @@
+package opslevel_jq_parser
+
+import (
+	"encoding/json"
+	"strings"
+
+	"github.com/rs/zerolog/log"
+
+	"github.com/opslevel/opslevel-go/v2024"
+)
+
+type JQPropertiesParser struct {
+	assigns []*JQFieldParser
+}
+
+func NewJQPropertiesParser(cfg PropertyRegistrationConfig) *JQPropertiesParser {
+	assigns := make([]*JQFieldParser, len(cfg.Assign))
+	for i, expression := range cfg.Assign {
+		assigns[i] = NewJQFieldParser(expression)
+	}
+	return &JQPropertiesParser{
+		assigns: assigns,
+	}
+}
+
+func propertyAssignmentFromMap(definition string, input map[string]interface{}) (opslevel.PropertyInput, error) {
+	var owner string
+	var value opslevel.JsonString
+	log.Debug().Msgf("property assignment '%s' input is %#v", definition, input)
+	for k, v := range input {
+		switch k {
+		case "owner":
+			owner = v.(string)
+		case "value":
+			jsonString, err := opslevel.NewJSONInput(v)
+			if err != nil {
+				return opslevel.PropertyInput{}, err
+			}
+			value = *jsonString
+		}
+	}
+	if owner != "" && value != "" {
+		return opslevel.PropertyInput{
+			Owner:      *opslevel.NewIdentifier(owner),
```

Comment:
no this doesn't come from the property - this should be the service and this struct should be built up in `kubectl-opslevel`

All the parser should return is a bunch of `key:value` for the prop def alias and the value to assign.

Reply:
Updated this to return a `key:value` map (`map[string]opslevel.JsonString`).

Key is the prop def alias and value is the JsonString.